src/pages/index.vue

```diff
@@ -42,72 +46,67 @@ const props = defineProps({
         default: ""
     }
 });
+type PropName = keyof typeof props;
 
 const indicatorNames = computed(() => (appConfig.value ? Object.keys(appConfig.value.indicators) : {}));
 const unknownProps: Ref<string[]> = ref([]);
 
-const notFoundMsg = (valueType, value) => `Unknown ${valueType}: ${value}.`;
+const notFoundMsg = (valueType: string, value: string) => `Unknown ${valueType}: ${value}.`;
 
 const notFoundDetail = computed(() => {
     return unknownProps.value.map((propName) => notFoundMsg(propName, props[propName])).join(" ");
 });
 
-const countryToSelect: Ref<null | string> = ref(null);
+const checkRouteProp = (propName: PropName, candidates: string[]) => {
+    // Do case-insensitive check against route prop - add to unknown props if not found
+    if (props[propName] === "") return "";
+    const pattern = new RegExp(`^${props[propName]}$`, "i");
+    return candidates.find((i) => pattern.test(i));
+};
 
 const selectDataForRoute = async () => {
-    if (!appConfig.value) {
-        return;
-    }
+    if (!appConfig.value) return;
 
     const unknown = [];
-    const checkRouteProp = (propName: "pathogen" | "version" | "indicator" | "country", candidates: string[]) => {
-        if (!props[propName]) {
-            return "";
-        }
-
-        // Do case-insensitive check against route prop - add to unknown props if not found
-        const pattern = new RegExp(`^${props[propName]}$`, "i");
-        const result = candidates.find((i) => pattern.test(i));
-        if (!result) {
-            unknown.push(propName);
-        }
-        return result;
+    const possibleValuesForProps: Record<PropName, string[]> = {
+        pathogen: [PATHOGEN],
+        version: [VERSION],
+        indicator: Object.keys(appConfig.value.indicators),
+        country: appConfig.value.countries
```

Comment:
i had to change a bit of this function for the refactor but i found the implementation confusing as i was changing the relevant bits of the refactor so thought i should just change the whole function because i actually think the function was doing something incorrectly anyway, i know we dont have different diseases/versions yet but this function would not have worked for those cases